### PR TITLE
[RA2 Ch3] Operator Pattern

### PR DESCRIPTION
**Pull request by petorre**
From @arkadykanevsky and me towards closing #2255.

### CODE REVIEW

**Review by arkadykanevsky — APPROVED** [review]

**Comment by rgstori** [comment]
can we highlight that operators are one (of many) way(s) to implement controllers for custom resources please? we can link to the CR chapter as well

**Comment by petorre** [comment]
> can we highlight that operators are one (of many) way(s) to implement controllers for custom resources please? we can link to the CR chapter as well

@rgstori : Yes, great suggestions. Please see how it is now written in commit [560d920](https://github.com/cntt-n/CNTT/pull/2416/commits/560d920f634d8ef5f1e41cba62d7e77b35636810).

**Review by CsatariGergely — APPROVED** [review]

**Comment by rgstori** [comment]
ok, pending merge of #2422

**Review by CsatariGergely — APPROVED** [review]

File: doc/ref_arch/kubernetes/chapters/chapter03.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
     * [3.2.2 Container Networking Services](#322-container-networking-services)
     * [3.2.3 Container Storage Services](#323-container-storage-services)
     * [3.2.4 Container Package Managers](#324-container-package-managers)
+        * [3.2.5.1 Operator Pattern](#3251-operator-pattern)
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.md:22]: Typo: 3.2.4.1 instead of 3.2.5.1

File: doc/ref_arch/kubernetes/chapters/chapter03.md [hunk 2]
```diff
@@ -465,3 +466,16 @@ complies with the CNCF Conformance test for the package managers to use in the
 lifecycle management of the applications they manage. The Reference Architecture
 does not recommend the usage of a Kubernetes Application package manager with a
 server side component installed to the Kubernetes Cluster (e.g.: Tiller).
+
+
+#### 3.2.5.1 Operator Pattern
+
+[Custom controller](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#custom-controllers) is a control loop watching cluster for changes to a custom resource. This tries to keep the current state of the resource in sync with the desired state.
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.md:473]: Suggestions
```suggestion
A [custom controller](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#custom-controllers) is a control loop that watches a custom resource for changes and tries to keep the current state of the resource in sync with the desired state.
```

File: doc/ref_arch/kubernetes/chapters/chapter03.md [hunk 3]
```diff
@@ -465,3 +466,16 @@ complies with the CNCF Conformance test for the package managers to use in the
...
+
+[Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) combines custom resources and custom controllers. Operators are software extensions to Kubernetes that capture operational knowledge and automate usage of custom resources to manage applications, their components and cloud infrastructure. 
+Operators can have different capability levels. As per repository [OperatorHub.io](https://operatorhub.io/), operator can have different capability levels ([picture](https://operatorhub.io/static/images/capability-level-diagram.svg)):
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter03.md:476]: typo:
```suggestion
Operators can have different capability levels. As per repository [OperatorHub.io](https://operatorhub.io/), an operator can have different capability levels ([picture](https://operatorhub.io/static/images/capability-level-diagram.svg)):
```

File: doc/ref_arch/kubernetes/chapters/chapter03.md (hunk 1 above)
Comment (petorre) [reply doc/ref_arch/kubernetes/chapters/chapter03.md]: @pgoyal01 : Actually it doesn't logically fit under 3.2.4 so I changed it into its own 3.2.5.

**Review by walterkozlowski — APPROVED** [review]
@petorre, understand that we have a resolution, hence I approve and I am happy to merge.  It looks like a file conflict, can you check please?

**Review by rgstori — APPROVED** [review]

**Comment by rgstori** [comment]
solved the conflict with custom resources (added this as a subsection), will merge now

**Pull request merged**